Retrofit rpm patches

stormi opened the pull request.

benjamreis commented on drivers/linstorvolumemanager.py:
```diff
@@ -278,6 +283,27 @@ def metadata(self, metadata):
+                if report.ret_code & linstor.consts.WARN_NOT_CONNECTED == \
+                        linstor.consts.WARN_NOT_CONNECTED:
+                    disconnected_hosts.add(pool.node_name)
+                    continue
```
Is the `continue` necessary here since it's the end of the loop?

Wescoeur replied:
It's normally a `break`. I fix it. :)

benjamreis commented on drivers/LinstorSR.py:
```diff
@@ -943,12 +984,60 @@ def _update_device_name(self, device_name):
+        # 5. Update sm config.
+        snap_vdi.sm_config = {}
+        snap_vdi.sm_config['vdi_type'] = vhdutil.VDI_TYPE_VHD
+        snap_vdi.sm_config['vhd-parent'] = snap_parent
```
```py
snap_vdi.sm_config = {
    'vdi_type': vhdutil.VDI_TYPE_VHD,
    'vhd-parent': snap_parent,
}
```

Also why `vdi_type` uses underscore and `vhd-parent` uses dash?

Wescoeur replied:
Yeah it's not very elegant... The names are already defined like this in the other drivers, for the moment I'm not sure it's a good idea to change that.

benjamreis replied:
Indeed let's not introduce too much differences in this PR (not its point).
Perhaps we'll unify that later in another one. :)

Wescoeur commented:
I close this useless PR, all changes are now in https://github.com/xcp-ng/sm/pull/2.

stormi committed:
Update xs-sm.service's description for XCP-ng

This was a patch added to the sm RPM git repo before we had this
forked git repo for sm in the xcp-ng github organisation.

stormi committed:
Add TrueNAS multipath config

This was a patch added to the sm RPM git repo before we had this
forked git repo for sm in the xcp-ng github organisation.

Wescoeur commented:
Ok. :+1:
